flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/LaunchableMesosWorker.java

```diff
@@ -332,6 +334,22 @@ public String toString() {
 		return taskInfo.build();
 	}
 
+	/**
+	 * Get port keys representing the TM's configured endpoints. This includes mandatory TM endpoints such as
+	 * data and rpc as well as optionally configured endpoints for services such as prometheus reporter
+	 *
+	 * @return A deterministicly ordered Set of port keys to expose from the TM container
+	 */
+	private Set<String> getPortKeys() {
+		LinkedHashSet<String> tmPortKeys = new LinkedHashSet<>(Arrays.asList(TM_PORT_KEYS));
+		containerSpec.getDynamicConfiguration().keySet().stream()
+			.filter(key -> key.endsWith(".port"))  // This matches property naming convention
```

Comment:
there are at least 2 instances where the key ends with "port**s**".
https://ci.apache.org/projects/flink/flink-docs-master/ops/config.html#query-proxy-ports
https://ci.apache.org/projects/flink/flink-docs-master/ops/config.html#query-server-ports